src/protocols/secure_channel/CASESession.cpp

```diff
@@ -178,24 +191,29 @@ CHIP_ERROR CASESession::ToSerializable(CASESessionSerializable & serializable)
 CHIP_ERROR CASESession::FromSerializable(const CASESessionSerializable & serializable)
 {
     mPairingComplete = (serializable.mPairingComplete == 1);
-    ReturnErrorOnFailure(mSharedSecret.SetLength(static_cast<size_t>(serializable.mSharedSecretLen)));
+    uint16_t length  = LittleEndian::HostSwap16(serializable.mSharedSecretLen);
+    ReturnErrorOnFailure(mSharedSecret.SetLength(static_cast<size_t>(length)));
+    memset(mSharedSecret, 0, sizeof(mSharedSecret.Capacity()));
+    memcpy(mSharedSecret, serializable.mSharedSecret, length);
 
-    VerifyOrReturnError(serializable.mMessageDigestLen <= sizeof(mMessageDigest), CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(serializable.mIPKLen <= sizeof(mIPK), CHIP_ERROR_INVALID_ARGUMENT);
+    length = LittleEndian::HostSwap16(serializable.mMessageDigestLen);
+    VerifyOrReturnError(length <= sizeof(mMessageDigest), CHIP_ERROR_INVALID_ARGUMENT);
+    memcpy(mMessageDigest, serializable.mMessageDigest, length);
 
-    memset(mSharedSecret, 0, sizeof(mSharedSecret.Capacity()));
-    memcpy(mSharedSecret, serializable.mSharedSecret, mSharedSecret.Length());
-    memcpy(mMessageDigest, serializable.mMessageDigest, serializable.mMessageDigestLen);
-    memcpy(mIPK, serializable.mIPK, serializable.mIPKLen);
+    length = LittleEndian::HostSwap16(serializable.mIPKLen);
+    VerifyOrReturnError(length <= sizeof(mIPK), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
It has to be the exact length of IPK. Lower will load a garbage IPK.

Also, I don't think we should serialize IPK, since it may no longer be in force in the future.

Reply:
I tried removing it in the current PR, but looks like some unit test is dependent on it. I do agree that it should be removed.
I added a TODO for time being to get back to it in a follow on PR.

Reply:
OK, well let's still make sure the IPK is exactly the expected length.